### PR TITLE
Remove compression stat update from update script

### LINKED ISSUE #2859: Slow update to 2.0 if lots of compressed data
The update script for 2.0.0-rc3 -> 2.0.0-rc4 runs a query over all compressed data to get the get the row count, which is then inserted into metadata tables. The purpose of this is to ensure `approximate_row_count` returns the correct row count. The update script change was part of this PR: https://github.com/timescale/timescaledb/pull/2555

Running a `SELECT count()` over all the compressed chunks can be very costly, causing the update script to hang and potentially block a lot of other transactions. 

A potential fix is to simply not update the metadata in the update script, and instead give documentation on how to do it after the update.

### PR DESCRIPTION
**Pull request by mkindahl**
Updating the compression stats can potentially be a very big task
locking many tables. Instead of doing this in the update script, we
move it to a separate function that can be used after the update is
done.

Fixes #2859

### CODE REVIEW

File: sql/update_util.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    _timescaledb_catalog.chunk comp
+  WHERE
+    compressed_chunk_id IS NOT NULL LOOP
+      SELECT
+        format('%I.%I', schema_name, table_name) INTO comp_chunk
+      FROM
+        _timescaledb_catalog.chunk
+      WHERE
+        id = plain_chunk.compressed_chunk_id;
+      EXECUTE format('SELECT sum(_ts_meta_count), count(*) FROM %s', comp_chunk) INTO rowcount_pre, rowcount_post;
```
Comment (davidkohn88) [inline sql/update_util.sql:30]: I'd recommend that we do a commit after each chunk here, there's no reason for it to be transactional and no reason to have one big long-running txn.

**Review by davidkohn88 — COMMENTED** [review]
LGTM, added one comment about transactionality of the proc calls.

File: sql/update_util.sql (hunk 1 above)
Comment (mkindahl) [reply sql/update_util.sql:30]: Good point. However, it is only going to be used once after the update, so we will remove this function from the PR and just describe how to do it in the documentation.

Comment (davidkohn88) [reply sql/update_util.sql:30]: Hmmm...I think I'd keep the proc and have people run the proc in the documentation seems a lot simpler for the user. You can also check to see whether the chunk has nulls in the metadata table in the where clause here I think, that way it'll only run once.

**Review by pmwkaa — APPROVED** [review]

**Review by svenklemm — APPROVED** [review]

**Comment by davidkohn88** [comment]
(I'm not sure if you saw my comment but I did like having the proc around better than just telling people what to do in the docs, I think it makes it much easier for them to actually update this...I would also add a test in the where clause of the select in the proc to check that the chunk has null stats then you can ensure that it only runs once per chunk even if they run the proc multiple times)

**Comment by mkindahl** [comment]
> (I'm not sure if you saw my comment but I did like having the proc around better than just telling people what to do in the docs, I think it makes it much easier for them to actually update this...I would also add a test in the where clause of the select in the proc to check that the chunk has null stats then you can ensure that it only runs once per chunk even if they run the proc multiple times)

> (I'm not sure if you saw my comment but I did like having the proc around better than just telling people what to do in the docs, I think it makes it much easier for them to actually update this...I would also add a test in the where clause of the select in the proc to check that the chunk has null stats then you can ensure that it only runs once per chunk even if they run the proc multiple times)

Hi David! It was proposed but we are not sure that we want to keep a lot of code around that are only used once. Since this is a ship-stopper, I think we should focus on fixing the current issue with the code being executed in the update script (which is a ship-stopper) and take the discussion about how to support post-update stuff (which is not a ship-stopper) separately.

File: sql/update_util.sql (hunk 1 above)
Comment (erimatnor) [reply sql/update_util.sql]: FWIW, we decided to move the script snippet to docs and not include in the extension. In docs, people can also adapt it for their own purposes (and we can easily modify it over time if needed).
